[ROCm] enable additional inductor/dynamo UTs

jataylo opened the pull request:
Enables additional inductor UTs on ROCm and un skips outdated skips. 

I have also removed a group of failures in `test_torchinductor_opinfo` which are now passing for CUDA and ROCm

```
-    # The following 3 tests fail on CUDA with AssertionError: expected size 5==5, stride 5==1 at dim=0
-    # linalg._svd's return value has different strides on CUDA vs CPU which causes this
-    # In test_meta.py there is a mechanism to skipping strides checks for some ops
-    # (including _linalg_svd), possibly we should have something similar here
-    "linalg.cond": {f32, f64},
-    "linalg.svdvals": {f32, f64},
-    "linalg.matrix_rank": {f32, f64},
-    "linalg.svd": {f32, f64},
```

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @hongxiayang @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @anijain2305

jataylo committed:
[ROCm] enable additional inductor/dynamo UTs on ROCm

jataylo committed:
linting suggestions

jataylo commented on test/inductor/test_torchinductor_opinfo.py:
```diff
@@ -336,14 +332,6 @@ def process(device_type):
-    # linalg._svd's return value has different strides on CUDA vs CPU which causes this
-    # In test_meta.py there is a mechanism to skipping strides checks for some ops
-    # (including _linalg_svd), possibly we should have something similar here
-    "linalg.cond": {f32, f64},
```
On ROCm we no longer observe failures for these 4 tests so no longer need to be added as expected failures. If CUDA still fails we can conditionalise, but I would like to see if this is now passing on CUDA side also.

davidradl replied:
ok makes sense - thanks for the explanation

jataylo committed:
Add `test_batch_norm_2d` with dynamic shapes to skip list on ROCm

jataylo committed:
Linting fix

jataylo committed:
Additional lint fix

jataylo committed:
skip `test_batch_norm_2d_2` on ROCm

Failure observed after unskip attempt

```
FAILED [0.1361s] inductor/test_torchinductor.py::CudaTests::test_batch_norm_2d_2_cuda - torch._dynamo.exc.BackendCompilerFailed: backend='inductor' raised:
LoweringException: RuntimeError: Expected contiguous tensor, but got non-contiguous tensor for argument #1 'input' (while checking arguments for miopen_batch_norm)
```

jataylo committed:
Fixes

jataylo commented:
@pytorchbot rebase

jataylo commented:
@pytorchbot rebase

jataylo commented:
@pytorchbot rebase

jataylo commented:
Hi @malfet @kit1980  enabling/unskipping some UTs for ROCm, could we get your help to review, CI is all green.

malfet commented:
@pytorchbot merge
